libpromises/rlist.c

```diff
@@ -325,18 +325,56 @@ Rlist *RlistAppendRval(Rlist **start, Rval rval)
     return rp;
 }
 
-Rval RvalNew(const void *item, RvalType type)
+Rval RvalNewRewriter(const void *item, RvalType type, JsonElement *map)
 {
     switch (type)
     {
     case RVAL_TYPE_SCALAR:
-        return (Rval) { xstrdup(item), RVAL_TYPE_SCALAR };
+        if (NULL != map && JsonLength(map) > 0)
+        {
+            // TODO: replace with BufferSearchAndReplace when the
```

Comment:
The `string_replace` PR from https://dev.cfengine.com/issues/7346 has a `BufferSearchAndReplace` function I will use here, once that PR is merged.  So please consider these workarounds temporary.